backend/src/routes/group.ts

```diff
@@ -0,0 +1,146 @@
+import Router from '@koa/router';
+import Group from '../models/group';
+import User from '../models/user';
+
+const router = new Router({
+	prefix: '/groups',
+});
+
+router.get('/', async (ctx, next) => {
+	ctx.body = await Group.find();
+});
+
+// Finds group by handle.
+router.get('/:grouphandle', async (ctx, next) => {
+	const group = await Group.findOneBy({
+		handle: ctx.params.grouphandle,
+	});
+	console.log(group);
+	if (group === null) {
+		ctx.status = 404;
+		return;
+	}
+	ctx.body = group;
+});
+
+//Create a new group with required group handle and display name.
+router.put('/:grouphandle/:displayname/:creatorhandle', async (ctx, next) => {
+	// once auth is set up, fix this to use object id provided by auth info.
+	const user = await User.findOneBy({
+		handle: ctx.params.creatorhandle,
+	});
+	if (user === null) {
+		console.log('This user does not exist.');
+		ctx.status = 404;
+		return;
+	}
+	const group = new Group();
+	group.handle = ctx.params.grouphandle;
+	group.displayname = ctx.params.displayname;
+	//Create new user array for followers and members.
+	group.followers = [];
+	group.members = [user];
+	group.timeline = [];
+	await group.save();
+	ctx.body = group;
+});
+
+// Add user to a group.
+router.put('/member/:grouphandle/:userhandle', async (ctx, next) => {
```

Comment:
let's change this to just a single user joining a group (as themself). so you can just grab the userhandle from auth instead of passing it into the route params